core/src/main/java/org/apache/accumulo/core/data/ByteSequence.java

```diff
@@ -157,4 +185,85 @@ public int hashCode() {
     return hash;
   }
 
+  /**
+   * This method returns an empty immutable byte sequence. The same object is always returned.
+   *
+   * @since 3.1.0
+   */
+  public static ByteSequence of() {
+    return EmptyByteSequence.INSTANCE;
```

Comment:
I'm wondering if `EmptyByteSequence` should extend `ImmutableByteSequence`. This would allow you to change the return type of these `of` methods to `ImmutableByteSequence` for stronger typing.

Reply:
Currently ImmutableByteSequence is package private.  Was loosely following the pattern of the java `of` methods on collections where the concrete type they might return is not public.  Do you think ImmutableByteSequence should be made public?